docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+1. RBAC rules allow designated verbs (`get`, `list`, `create`, `update`, etc.)
+   on the specified API resources for the specified subjects. If a subject,
+   e.g., a `ServiceAccount`, is allowed to `create` a managed resource, RBAC
+   alone cannot be used to constrain the set of `ProviderConfig`s that can be
+   referenced by the `create`d managed resource.
+1. The tenant subject itself does not directly access the `ProviderConfig` and in turn
+   the Cloud provider credential set referred by the `ProviderConfig`. It's the
+   Crossplane provider's `ServiceAccount` that accesses these resources, and as
+   mentioned above, this ServiceAccount currently serves all tenants. This
+   implies that we cannot isolate Cloud provider credentials among namespaced
+   tenants by only using RBAC rules if we allow tenant subjects to have `edit`
+   access (`create`, `update`, `patch`) to managed resources. Although it's
+   possible to prevent them from reading Cloud provider credentials of other
+   tenants in the cluster via RBAC rules, it's not possible to prevent them from
+   _using_ those credentials solely with RBAC.
+
+As discussed in detail in the [Crossplane Multi-tenancy Guide][xp-mt],
+Crossplane is opinionated about the different personas in an organization adopting
+Crossplane. We make a distinction between the _infrastructure operators_ (or
+_platform builders_) who are expected to manage cluster-scoped resources (like
+`ProviderConfig`s, XRDs and `Composition`s) and _application operators_, who are
+expected to consume the infrastructure for their applications. And tenant
+subjects are classified as _application operators_, i.e., it's the
+infrastructure operator's responsibility to manage the infrastructure _across_
+the tenants via cluster-scoped Crossplane resources, and it's possible and
+desirable from an isolation perspective to disallow application operators, who
+are tenant subjects, to directly access these shared cluster-scoped resources.
+This distinction is currently possible with Crossplane because: 
+1. Crossplane `Claim` types are defined via cluster-scoped XRDs by
+   infrastructure operators and _namespaced_ `Claim` instances are used by the
+   tenant subjects. This allows infrastructure operators to define RBAC rules
+   that allow tenant subjects to only access resources in their respective
+   namespaces, e.g., `Claim`s.
+1. However, [1] is not sufficient on itself, as the scheme is still prone to
+   privilege escalation attacks if the API exposed by the XR is not well designed. The
+   (shared) provider `ServiceAccount` has access to all Cloud provider
+   credentials in the cluster and if the exposed XR API allows a `Claim` to
+   reference cross-tenant `ProviderConfig`s, then a misbehaving tenant subject
+   can `create` a `Claim` which references some other tenant's credential set.
+   Thus in our multi-tenancy [guide][xp-mt], we propose a security scheme where:
```

Comment:
The `[guide]` link is broken.

Reply:
Unfortunately can't find document link to resolve this.

Reply:
We can use the following here: https://docs.crossplane.io/knowledge-base/guides/multi-tenant/

Reply:
Fixed